test/python/circuit/test_controlled_gate.py

```diff
@@ -379,6 +379,13 @@ def test_inverse_circuit(self, num_ctrl_qubits):
         np.testing.assert_array_almost_equal(result.data,
                                              np.identity(result.dim[0]))
 
+    def test_named_circuit(self):
+        """Tests control is applied to operation type, not name"""
+        qc = QuantumCircuit(1, name='x')
+        gate = qc.to_gate()
+
+        self.assertIsNone(gate.control().definition)  # Not a CnotGate
```

Comment:
@ewinston looks like a change in https://github.com/Qiskit/qiskit-terra/pull/3739, specifically here:
https://github.com/Qiskit/qiskit-terra/blob/025f6f9e73572a4aa608ac6e90bb1d7bc557e51f/qiskit/circuit/controlledgate.py#L85

is causing the definition get in this test to throw with
`TypeError: can only concatenate list (not "NoneType") to list`
because the gate generated from this circuit has no definition.

Is that ok/expected, since the gate here is degenerate? I originally added no operations to the test circuit, since it wouldn't be relevant to the testing the name bug, but I can add now if you think my assessment is accurate.

Reply:
I think in this special case, where the controlled gate has an empty `_definition`, the `definition` method of ControlledGate should perhaps also return the same. Could you add an
 ```
else:
     return self._definition
```
to see if that resolves it?